Gems/ROS2/Code/Source/RobotImporter/URDF/VisualsMaker.cpp

```diff
@@ -231,7 +232,54 @@ namespace ROS2
         entity->Deactivate();
     }
 
-    static void OverrideMaterialPbrSettings(const sdf::Material* material, [[maybe_unused]] AZ::Render::MaterialAssignmentMap& overrides)
+    static void OverrideScriptMaterial(const sdf::Material* material, [[maybe_unused]] AZ::Render::MaterialAssignmentMap& overrides)
+    {
+        AZStd::string materialName(material->ScriptName().c_str(), material->ScriptName().size());
+        if (materialName.empty())
+        {
+            return;
+        }
+
+        // If a material has a <script> element we'll treat the name as a path and name to an O3DE material.
+        // For example, "Gazebo/Wood" will look for a product material in "<cache>/gazebo/wood.azmaterial"
+        AZ::IO::Path materialProductPath(materialName);
+        materialProductPath.ReplaceExtension(".azmaterial");
```

Comment:
Shouldn't we lowercase the material name here?
```suggestion
        AZStd::string materialName(material->ScriptName().c_str(), material->ScriptName().size());
        if (materialName.empty())
        {
            return;
        }

        // Make sure the material name is lowercased before checking the path in the Asset Cache
        AZStd::to_lower(materialName);
        // If a material has a <script> element we'll treat the name as a path and name to an O3DE material.
        // For example, "Gazebo/Wood" will look for a product material in "<cache>/gazebo/wood.azmaterial"
        AZ::IO::Path materialProductPath(materialName);
        materialProductPath.ReplaceExtension(".azmaterial");
```

Reply:
It's not _required_, since the asset cache lookup with also to_lower it, but i'm great with making it explicit here.